infra/docker-images/db-ops/anonymize.sql

```diff
@@ -1,198 +1,58 @@
- RETURNS void
- LANGUAGE plpgsql
-AS $function$
-DECLARE
-    query text;
-BEGIN
-    EXECUTE 'ALTER TABLE ' || table_name || ' DISABLE TRIGGER audit_trigger_delete;';
-    EXECUTE 'ALTER TABLE ' || table_name || ' DISABLE TRIGGER audit_trigger_insert;';
-    EXECUTE 'ALTER TABLE ' || table_name || ' DISABLE TRIGGER audit_trigger_update;';
-END;
-$function$;
-
-CREATE OR REPLACE FUNCTION pg_temp.enable_activity_trigger(table_name text)
- RETURNS void
- LANGUAGE plpgsql
-AS $function$
-DECLARE
-    query text;
-BEGIN
-    EXECUTE 'ALTER TABLE ' || table_name || ' ENABLE TRIGGER audit_trigger_delete;';
-    EXECUTE 'ALTER TABLE ' || table_name || ' ENABLE TRIGGER audit_trigger_insert;';
-    EXECUTE 'ALTER TABLE ' || table_name || ' ENABLE TRIGGER audit_trigger_update;';
-END;
-$function$;
-
-ALTER TABLE "user" DISABLE TRIGGER audit_trigger_insert;
-ALTER TABLE "user" DISABLE TRIGGER audit_trigger_update;
-ALTER TABLE "user" DISABLE TRIGGER audit_trigger_delete;
-
-SELECT pg_temp.disable_activity_trigger('offerer');
-SELECT pg_temp.disable_activity_trigger('venue');
-SELECT pg_temp.disable_activity_trigger('booking');
-SELECT pg_temp.disable_activity_trigger('stock');
-SELECT pg_temp.disable_activity_trigger('bank_information');
-SELECT pg_temp.disable_activity_trigger('mediation');
-SELECT pg_temp.disable_activity_trigger('offer');
-SELECT pg_temp.disable_activity_trigger('product');
-SELECT pg_temp.disable_activity_trigger('venue_provider');
 
-UPDATE offerer SET "validationToken" = substring(md5(random()::text),1 , 27) WHERE "validationToken" is not null;
-
-UPDATE bank_information SET "iban" = pg_temp.generate_random_between(999999999,100000000)::text WHERE "iban" is not null;
-UPDATE bank_information SET "bic" = pg_temp.generate_random_between(999999999,100000000)::text WHERE "bic" is not null;
-
-UPDATE booking SET "token" = pg_temp.generate_booking_token_from_id("id") WHERE "token" is not null;
-
-UPDATE payment SET "iban" = 'FR7630001007941234567890185' WHERE "iban" is not null;
-UPDATE payment SET "bic" = 'BDFEFR2L' WHERE "bic" is not null;
-
-UPDATE provider SET "apiKey" = substring(md5(random()::text), 1, 32) WHERE "apiKey" is not null;
+UPDATE offerer SET "validationToken" = pg_temp.random_text(27) WHERE "validationToken" is not null;
+
+BEGIN;
+  -- Temporarily remove trigger to speed up updates.
+  DROP TRIGGER IF EXISTS stock_update_cancellation_date ON booking;
+  -- Temporarily set all tokens to NULL to avoiding triggering
+  -- UNIQUEness error on existing tokens on the second UPDATE.
+  ALTER TABLE booking ALTER COLUMN token DROP NOT NULL;
+  UPDATE booking SET token = NULL;
+  UPDATE booking SET token = pg_temp.generate_booking_token_from_id(id);
+  -- Restore constraints and triggers.
+  ALTER TABLE booking ALTER COLUMN token SET NOT NULL;
+  CREATE TRIGGER stock_update_cancellation_date
+    BEFORE INSERT OR UPDATE ON booking
+    FOR EACH ROW
+    EXECUTE PROCEDURE save_cancellation_date();
```

Comment:
```suggestion
  -- Temporarily disable trigger to speed up updates.
  ALTER TABLE booking DISABLE TRIGGER stock_update_cancellation_date;
  -- Temporarily set all tokens to NULL to avoiding triggering
  -- UNIQUEness error on existing tokens on the second UPDATE.
  ALTER TABLE booking ALTER COLUMN token DROP NOT NULL;
  UPDATE booking SET token = NULL;
  UPDATE booking SET token = pg_temp.generate_booking_token_from_id(id);
  -- Restore constraints and triggers.
  ALTER TABLE booking ALTER COLUMN token SET NOT NULL;
  ALTER TABLE booking ENABLE TRIGGER stock_update_cancellation_date;
```

Y a-t-il une différence de performance entre un drop/create trigger et disable/enable trigger ?

Reply:
Je ne connaissais pas (et n'ai pas pensé à chercher avant). C'est instantané et surtout c'est bien plus malin que de devoir maintenir à deux endroits ce `CREATE TRIGGER`. Merci @francois-seguin  ! :)